File: python/7 kyu/433.py
# https://www.codewars.com/kata/6347f9715467f0001b434936/train/python

# Passed

def thin_or_fat(matrix):
    rooted_widths_sum = 0
    rooted_heights_sum = 0
    for row in matrix:
        width_sum = sum(row)
        if width_sum < 0:
            return None

        rooted_widths_sum += width_sum ** 0.5
    
    for col_idx in range(len(matrix[0])):
        height_sum = sum(matrix[row_idx][col_idx] for row_idx in range(len(matrix)))
        if height_sum < 0:
            return None
        
        rooted_heights_sum +=  height_sum ** 0.5
        
    if rooted_widths_sum > rooted_heights_sum:
        result = "fat"
    elif rooted_widths_sum < rooted_heights_sum:
        result = "thin"
    else:
        result = "perfect"
        
    return result

output = thin_or_fat([[1,3], [5,7]])
print(output)

File: nodejs/8 kyu/433.js
// https://www.codewars.com/kata/57a083a57cb1f31db7000028/train/javascript

// Passed

function powersOfTwo(n) {
  let result = [];
  for (let pow = 0; pow <= n; pow++) {
    result.push(2 ** pow);
  }
  return result;
}

const output = powersOfTwo(4);
console.log(output);